rust/parquet/src/basic.rs

```diff
@@ -337,8 +396,11 @@ impl ColumnOrder {
             // If the max is -0, the row group may contain +0 values as well.
             // When looking for NaN values, min and max should be ignored.
             Type::FLOAT | Type::DOUBLE => SortOrder::SIGNED,
-            // unsigned byte-wise comparison
-            Type::BYTE_ARRAY | Type::FIXED_LEN_BYTE_ARRAY => SortOrder::UNSIGNED,
+            // Unsigned byte-wise comparison
+            Type::BYTE_ARRAY => SortOrder::UNSIGNED,
+            // Only unsigned if there was a logical type that supports unsigned sort.
+            // Interval has no defined sort order, and should not use UNSIGNED.
```

Comment:
Not quite sure about this: since `INTERVAL` is already handled above with `UNDEFINED`, why should we use `UNDEFINED` for this? 

Seems [parquet-mr](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveType.java#L507) only uses `UNDEFINED` for `INT96` and `INTERVAL`.

Reply:
After thinking more about it, I agree with you. I initially resorted to only using LogicalType + PhysicalType, in which case there would have been no ConvertedType covering INTERVAL. I changed this after realising that it was a mistake; but forgot to revise this.

I'll change this.